src/wasm_plugin.rs

```diff
@@ -32,7 +32,7 @@ impl SyncPluginHandler<Configuration> for DockerfilePluginHandler {
       },
       file_matching: FileMatchingInfo {
         file_extensions: vec!["dockerfile".to_string()],
-        file_names: vec!["Dockerfile".to_string()],
+        file_names: vec!["Dockerfile".to_string(), "Dockerfile.*".to_string()],
```

Comment:
This is arguably a bit of a hack/abuse (I have a feeling that `file_names` was not intended to be used with globs in the entries), but looking at dprint main sources, I think it would work. Setting up `file_prefixes` for something like this would sound a bit overkill to me.

Reply:
Yeah, it won't work. This doesn't do pattern matching.

I think the only way to make this work in dprint is for someone to manually add "associations" (https://dprint.dev/config/#associations) under their dockerfile config.

Overall, I'm not too interested in suporting the `dockerfile.*` convention because it's a bad convention and it's good the project has switched over to using a file extension.

Reply:
For reference, I think this will work today:

```
{
  "dockerfile": {
    "associations": [
      "Dockerfile.*",
      "Dockerfile",
      "*.dockerfile"
    ]
  },
  "plugins": [
    "https://plugins.dprint.dev/dockerfile-0.3.2.wasm"
  ]
}
```

Reply:
The reason I think it could work is that `file_names` entries seem to be just concatenated to `**/` and it doesn't seem there's anything magical whether the entries are globs or plain filenames: https://github.com/dprint/dprint/blob/14eb074e9baf8d0533f04ad954e17a5a375e6385/crates/dprint/src/paths.rs#L161

Using "associations" works, I'm using that currently, but it's a chore that one has to repeat all the built in globs in it.

Reply:
Oh, yeah, maybe it will work. You can try building by calling:

https://github.com/dprint/dprint-plugin-dockerfile/blob/d208687d1090f99b114f3a65e240460f49c4f315/.github/workflows/ci.yml#L37

Then the wasm file will be somewhere in the `target` folder (just import the file relatively from your dprint.config file)

Reply:
Hmph, doesn't seem to work for some reason after all :/

Reply:
It expands to this glob:

```
**/{Dockerfile,Dockerfile.*}
```

I guess that doesn't work. Maybe the dprint cli could be updated to support some basic wildcards in filenames to make it work.